### PR TITLE
Added legacy test 001

### PR DESCRIPTION
**Pull request by akelson**
Because Neovim always ships with all features the small.vim, tiny.vim, mbyte.vim, mzscheme.vim, and lua.vim files are not applicable. It is also useless to create 'wrontermsize' since the existence of this file was checked by the legacy test makefile. Instead a minimum terminal size assert is performed.

### CODE REVIEW

**Comment by akelson** [comment]
What does it take to get a pull request to be acknowledged?

**Comment by justinmk** [comment]
acknowledged, and appreciated. But see https://github.com/neovim/neovim/pull/1462 for previous discussion of a similar PR. Looks like you actually have taken into consideration @elmart's concern from that discussion.

File: test/functional/legacy/001_script_test_spec.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+local eval = helpers.eval
+
+describe('test script', function()
+  setup(clear)
+
+  it('is working', function()
+    -- Write a single line to test.out to check if testing works at all.
+    execute('%d')
+    feed('athis is a test<esc>:w! test.out<cr>')
```
Comment (justinmk) [inline test/functional/legacy/001_script_test_spec.lua]: AFAIK writing `test.out` is not necessary in the migrated functional tests. Is test001 a special case where it's needed?

compare https://github.com/neovim/neovim/pull/1462/files

Comment (justinmk) [inline test/functional/legacy/001_script_test_spec.lua]: If it makes sense to do so, just remove `:w! test.out<cr>` and then this LGTM

**Comment by elmart** [comment]
> What does it take to get a pull request to be acknowledged?

It seems you've hit a low-activity period of those people acquainted with test-related stuff. Don't worry. Your PR will be eventually accepted, if useful.

Have you seen my comment about terminal size mentioned by @justinmk? What do you think of that?

Also, I think your two commits should be squashed into one.

File: test/functional/legacy/001_script_test_spec.lua (hunk 1 above)
Comment (akelson) [inline test/functional/legacy/001_script_test_spec.lua]: Thank you for responding. This is my first experience with open source development or github so please excuse my ignorance. Yes, it looks like you are right. The `:w! test.out<cr>` should be removed.

**Comment by akelson** [comment]
Thanks for the replies! 

I looked at your comment, @elmart, and agree that test 1 should be left in the old test suite. Would it be acceptable to migrate test 1 to lua/busted now and remove the old test later? I also agree that a check of the work area size should be retained in the new tests. I want to do a little more investigation to understand where the default size comes from when running in headless mode.

**Comment by justinmk** [comment]
> Would it be acceptable to migrate test 1 to lua/busted now and remove the old test later? 

:+1: 

> how to combine my commits.

`git rebase -i upstream/master`

File: test/functional/legacy/001_script_test_spec.lua [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    expect([[
+      this is a test]])
+  end)
+end)
+
+describe('term size', function()
+  setup(clear)
+
+  it('is greater than 80x24', function()
+    -- (Some tests will fail. When columns and/or lines are small).
```
Comment (justinmk) [inline test/functional/legacy/001_script_test_spec.lua]: the auto-migration script was too eager here. This is one sentence, not two.

File: test/functional/legacy/001_script_test_spec.lua [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+local eval = helpers.eval
+
+describe('test script', function()
+  setup(clear)
+
+  it('is working', function()
+    -- Write a single line to test.out to check if testing works at all.
```
Comment (justinmk) [inline test/functional/legacy/001_script_test_spec.lua]: This comment is confusing, we don't write test.out in the new tests. Suggestion:

    -- Write a single line to check if testing works at all.

File: test/functional/legacy/001_script_test_spec.lua [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+describe('term size', function()
+  setup(clear)
+
+  it('is greater than 80x24', function()
+    -- (Some tests will fail. When columns and/or lines are small).
+    local lines = eval('&lines')
+    local columns = eval('&columns')
+    assert.False( lines < 24 )
+    assert.False( columns < 80 )
```
Comment (justinmk) [inline test/functional/legacy/001_script_test_spec.lua]: Wouldn't `assert.True( foo >= bar )` read more naturally? Usually we prefer "positive logic".

File: test/functional/legacy/001_script_test_spec.lua [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    -- Assert buffer contents.
+    expect([[
+      this is a test]])
+  end)
+end)
+
+describe('term size', function()
+  setup(clear)
+
+  it('is greater than 80x24', function()
```
Comment (justinmk) [inline test/functional/legacy/001_script_test_spec.lua]: should actually read "at least 80x24", since that is what the test actually tests.

**Comment by akelson** [comment]
Good catches, @justinmk. I agree with all of your comments and will make the changes. Should I squash the commits so that there is just one commit on this pull request when I'm done?

**Comment by akelson** [comment]
I asked this on IRC and didn't get a response, so I'll ask here. When adding tests is it best to make a pull request for each new test, or would it be best to add several tests under one pull request?

**Comment by justinmk** [comment]
Yes please squash.

>  is it best to make a pull request for each new test, or would it be best to add several tests under one pull request?

Grouping is a matter of judgment but to answer your specific question, it's fine to group several tests in one PR. Avoid huge PRs because they take longer to process/rework; but may tiny PRs may scatter discussion and communication.

**Comment by justinmk** [comment]
@akelson Would you mind signing the CLA? https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#pull-requests

File: test/functional/legacy/001_script_test_spec.lua (hunk 3 above)
Comment (akelson) [inline test/functional/legacy/001_script_test_spec.lua]: Will change to:
`-- Insert a single line to check if testing works at all.`

**Comment by justinmk** [comment]
Thanks. LGTM after squash

**Comment by tarruda** [comment]
@elmart 's quote in #1462 

> Original test1 checks for terminal size to be adequate for other tests to run. If I'm not wrong, it's kind of a precondition for the old test suite.
So, IMO, it shouldn't be removed until all other tests have dissapeared.

Thats 100% true, so there's no need to migrate this test.

> Now, in addition to that, there's something I'm not sure about:
Translated test doesn't check terminal size. That's ok since we don't have a terminal there. But, shouldn't we otherwise checking work area size? I mean, some test results will depend on how long lines are (think about wrapping, formatting, etc), which depend on size. So, even in headless mode, we should a have a size concept somewhere, shouldn't we?

We do have this concept, and the default is already hardcoded in globals.h to 80x24. Resizing in headless mode is done by attaching a UI instance and there are some real screen testing examples [here](https://github.com/neovim/neovim/tree/master/test/functional/ui). So, for  tests that dont attach a UI there's no need to check this because the screen is always 80x24 and is reset after each test.

@akelson I'm sorry it took so long for me to look at this PR again, but after my first visit to this PR I thought it was already clear from #1462 that test 1 is not a real test and doesn't need to be migrated to the new infrastructure

**Comment by justinmk** [comment]
@tarruda I thought it might be ok to have this test as a sanity check, but you make a strong case, so I will close it. @akelson I hope at least this PR was useful as an exercise. If you are willing, there should be other moderately sized tests still in need of migration.

**Comment by tarruda** [comment]
> @tarruda I thought it might be ok to have this test as a sanity check, but you make a strong case, so I will close it. @akelson I hope at least this PR was useful as an exercise. If you are willing, there should be other moderately sized tests still in need of migration.

Indeed. Tests that depend on screen size should be migrated to screen.lua, which is another good exercise.

**Pull request closed without merging**